src/main/AppConnector.h

```diff
@@ -10,25 +11,35 @@ class LedgerManager;
 class Herder;
 class BanManager;
 struct OverlayMetrics;
+class SorobanNetworkConfig;
+class SorobanMetrics;
+struct LedgerTxnDelta;
 
 // Helper class to isolate access to Application; all function helpers must
 // either be called from main or be thread-sade
-class OverlayAppConnector
+class AppConnector
 {
     Application& mApp;
     // Copy config for threads to use, and avoid warnings from thread sanitizer
     // about accessing mApp
     Config const mConfig;
 
   public:
-    OverlayAppConnector(Application& app);
+    AppConnector(Application& app);
 
     // Methods that can only be called from main thread
     Herder& getHerder();
     LedgerManager& getLedgerManager();
     OverlayManager& getOverlayManager();
     BanManager& getBanManager();
     bool shouldYield() const;
+    SorobanNetworkConfig const& getSorobanNetworkConfig() const;
+    medida::MetricsRegistry& getMetrics() const;
+    SorobanMetrics& getSorobanMetrics() const;
+    void checkOnOperationApply(Operation const& operation,
+                               OperationResult const& opres,
+                               LedgerTxnDelta const& ltxDelta);
+    Hash const& getNetworkID() const;
```

Comment:
I think removing getNetworkID completely out of App would be a bit annoying, since it's a pretty invasive change and we'd need to switch the rest of the codebase to using AppConnector, which doesn't seem right (since not everyone needs thread-safety).

In the changes that use this method from the background, we'd need to rework is a bit to ensure thread-safety (it is technically safe already, but need to prevent tsan warnings). We can then rename it to something like `getNetworkIDThreadSafe` to add clarity.